fairmq/FairMQMessage.h

```diff
@@ -29,15 +29,15 @@ class FairMQMessage
     virtual void Rebuild(const size_t size) = 0;
     virtual void Rebuild(void* data, const size_t size, fairmq_free_fn* ffn, void* hint = nullptr) = 0;
 
-    virtual void* GetMessage() = 0;
-    virtual void* GetData() = 0;
-    virtual size_t GetSize() = 0;
+    virtual void* GetData() const = 0;
+    virtual size_t GetSize() const = 0;
 
-    virtual void SetMessage(void* data, size_t size) = 0;
+    virtual bool SetUsedSize(const size_t size) = 0;
 
     virtual FairMQ::Transport GetType() const = 0;
 
-    virtual void Copy(const std::unique_ptr<FairMQMessage>& msg) = 0;
+    virtual void Copy(const std::unique_ptr<FairMQMessage>& msg) __attribute__((deprecated("Use 'Copy(const FairMQMessage& msg)'"))) = 0;
```

Comment:
I think its difficult to force such a hard API change. Can the old API be supported for a while? There is anyhow no ownership transferred here, so the unique pointer can be dereferenced and the new API be called.

Reply:
Why is it problematic?
It is not really forced, since it is just marked as deprecated and the old version works (at least for a while). The warning should help you switch to new API.
Cannot imagine an easier way to change API.

Reply:
In the present case, the API change breaks the O2 CI, and there is no way to fix this consistently, because the CI does checks with both the dev and the master, but only one version of the API is allowed.

Reply:
We will keep the old version at least until the end of next release. You can switch to the new API once the master is updated, there is no rush to change this now in your code.

Or how else would you do it?

Reply:
true, then everything is correct, actually the CI is broken because we treat the warnings as errors. Thanks for the clarification.

Somehow I was expecting the wrapper for the old API to be in the header file itself, which made me confused.